pkg/apis/monitoring/v1/types.go

```diff
@@ -1015,3 +1015,55 @@ const (
 	SelectorMechanismRelabel SelectorMechanism = "RelabelConfig"
 	SelectorMechanismRole    SelectorMechanism = "RoleSelector"
 )
+
+// WorkloadResourceReference is a link of workload resource wuth configuration resource.
+// +k8s:openapi-gen=true
+type WorkloadResourceReference struct {
+	// The group of the referenced resource.
+	// +kubebuilder:validation:Enum=monitoring.coreos.com
+	// +required
+	Group string `json:"group"`
+	// The type of resource being referenced (e.g. Prometheus or PrometheusAgent).
+	// +kubebuilder:validation:Enum=prometheuses;prometheusagents
+	// +required
+	Resource string `json:"resource"`
+	// The name of the referenced object.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name"`
+	// The namespace of the referenced object.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Namespace string `json:"namespace"`
+	// The current state of the service monitor when bound to the referenced Prometheus object.
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []ConfigResCondition `json:"conditions,omitempty"`
```

Comment:
I think I might be a bit out of the loop for this. But what I gather is: 
smon's status --> resources that select it --> contains the status of other config resources? 

This seems a bit confusing to me. Why can't we have the `Conditions` field be part of the resources themselves?